duckduckgo/Android
# Add capability to download blobs

Task/Issue URL: https://app.asana.com/0/414730916066338/1199183428824728
Tech Design URL: 
CC: 

**Description**:
This PR adds the capabilities to download blobs (albeit not being able to respect the filename). It fixes https://github.com/duckduckgo/Android/issues/1010

**Steps to test this PR**:
1. Go to https://eligrey.com/demos/FileSaver.js/
1. Try to download the three types of files on that site.
1. All those files should successfully download and opened within the mobile device.

**Other test**
1. Go to https://media.interieur.gouv.fr/deplacement-covid-19/
1. Fill the form and tap on "Générer mon attestation"
1. The pdf file should be downloaded successfully

**Failure**
1. Edit the js file to return `BlobConverter.convertBlobToDataUri('error', '%contentType%');` in the line 32.
1. Go to https://eligrey.com/demos/FileSaver.js/
1. Try to download any file from there.
1. You should see an error message.

**Timeout/Killing the webview**
1. Edit the line 32 in the js file to execute `setTimeout(function(){ BlobConverter.convertBlobToDataUri(dataUrl, '%contentType%'); }, 5000);`. This should delay the callback to the app by 5 seconds.
1. Kill the app or navigate to a different site or close the tab
1. The app should not crash.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
